supervisor/snapshots/__init__.py

```diff
@@ -297,6 +302,7 @@ async def do_restore_partial(
                 if FOLDER_HOMEASSISTANT in folders:
                     await self.sys_homeassistant.core.stop()
                     snapshot.restore_homeassistant()
+                    snapshot.restore_dockerconfig()
```

Comment:
Why only restore this `if FOLDER_HOMEASSISTANT in folders` ?

Reply:
Should this be restored unconditionally? This is not a folder, but a file, so I can't just pass it to `folders` I guess. I could add a new attribute that would list all the configfiles we want to restore (and back up), but I am afraid that would be a more invasive change than this whole PR itself.

Reply:
Yeah, we should always store and restore this configuration, and the restoration of it needs to happen before the restoration of anything else container-related.